lib/spree/backend/engine.rb

```diff
@@ -3,10 +3,13 @@
 module Spree
   module Backend
     class Engine < ::Rails::Engine
+      Environment = Struct.new(:main_menu)
+
       config.middleware.use 'Spree::Backend::Middleware::SeoAssist'
 
-      initializer 'spree.backend.environment', before: :load_config_initializers do |_app|
+      initializer 'spree.backend.environment', before: :load_config_initializers do |app|
         Spree::Backend::Config = Spree::Backend::Configuration.new
+        app.config.spree_backend = Environment.new
```

Comment:
1. I think "Environment" is not a great name in this context.
2. Isn't it a little awkward that now there are two backend "configs"? The Preference-based `Spree::Backend::Config` and this `Environment` struct.